src/clj/harja/kyselyt/suolarajoitus_kyselyt.sql

```diff
@@ -122,90 +124,58 @@ WHERE ra.poistettu = FALSE
+  AND sr.materiaalikoodi = mk.id
+GROUP BY pvm, sr.materiaalikoodi, mk.nimi, k.jarjestelma
 UNION
-SELECT mk.id                          AS materiaali_id,
+SELECT sr.materiaalikoodi             AS materiaali_id,
        mk.nimi                        AS "materiaali-nimi",
        date_trunc('day', tot.alkanut) AS pvm,
        null                           AS suolamaara,
-       SUM(mat.maara)                 AS formiaattimaara,
+       SUM(sr.maara)                  AS formiaattimaara,
        null                           AS suolalukumaara,
-       count(mat.maara)               AS lukumaara,
-       TRUE                           AS koneellinen
+       count(sr.maara)                AS formiaattilukumaara,
+       k.jarjestelma                  AS "koneellinen?"
 FROM toteuma tot
-         LEFT JOIN toteuman_reittipisteet tr ON tr.toteuma = tot.id -- Rajoitetaan nämä vain formiaatteihin
-         LEFT JOIN LATERAL unnest(tr.reittipisteet) AS trp ON TRUE AND trp.aika BETWEEN :alkupvm::DATE - INTERVAL '1 day' AND :loppupvm::DATE
-         LEFT JOIN LATERAL unnest(trp.materiaalit) as mat ON TRUE AND mat.materiaalikoodi in (SELECT id FROM formiaatit),
-     materiaalikoodi mk,
-     rajoitusalue ra
-WHERE ra.id = :rajoitusalue-id
-  AND mat.materiaalikoodi = mk.id
-  AND tot.poistettu = FALSE
+     JOIN suolatoteuma_reittipiste sr ON sr.toteuma = tot.id AND sr.rajoitusalue_id = :rajoitusalue-id
+     JOIN materiaalikoodi mk ON mk.id = sr.materiaalikoodi AND mk.materiaalityyppi = 'formiaatti'
+     JOIN kayttaja k ON tot.luoja = k.id
+WHERE tot.poistettu = FALSE
   AND tot.urakka = :urakka-id
-  AND ST_DWithin(ra.sijainti, trp.sijainti::geometry, 1)
   AND tot.alkanut BETWEEN :alkupvm::DATE - INTERVAL '1 day' AND :loppupvm::DATE
-GROUP BY pvm, mk.id
+  AND sr.aika BETWEEN :alkupvm::DATE AND :loppupvm::DATE
+  AND sr.materiaalikoodi = mk.id
+GROUP BY pvm, sr.materiaalikoodi, mk.nimi, k.jarjestelma
 ORDER BY pvm ASC, "materiaali-nimi" ASC;
 
 -- name: hae-rajoitusalueen-paivan-toteumat
-WITH rageom AS (
-    SELECT sijainti
-    FROM rajoitusalue ra
-    WHERE ra.id = :rajoitusalue-id AND ra.poistettu = FALSE
-),
- formiaatit AS (
-     SELECT id FROM materiaalikoodi WHERE materiaalityyppi = 'formiaatti'::materiaalityyppi
- )
 SELECT tot.id AS id,
        tot.alkanut AS alkanut,
        tot.paattynyt AS paattynyt,
-       SUM(rp.maara) as suolamaara,
-       null as formiaattimaara
+       SUM(rp.maara) as maara,
+       concat(tot.tyokonetyyppi, ', ', tot.tyokonetunniste, ', ', tot.tyokoneen_lisatieto) as lisatieto
 FROM toteuma tot
-         LEFT JOIN suolatoteuma_reittipiste rp
+     JOIN suolatoteuma_reittipiste rp
              ON rp.toteuma = tot.id
-                AND rp.materiaalikoodi = :materiaali-id,
-     rageom
+                AND rp.materiaalikoodi = :materiaali-id
+                AND rp.rajoitusalue_id = :rajoitusalue-id
+     JOIN kayttaja k ON tot.luoja = k.id AND k.jarjestelma = :koneellinen?
```

Comment:
voiko :koneellinen? olla null? Mitä tapahtuu jos on?

Reply:
Teoriassa se voidaan asettaa nulliksi, koska Not Null määrettä ei ole pakotettu, mutta default on false ja tuotannossa ei ole yhtään nullia arvoa. Eli ei voi olla null.

Reply:
Aivan! En hoksannut, kuinka se päätyy frontilta tänne, mutta nyt hoksasin missä se tieto kulkee :+1: